File: back/src/main/java/br/eng/eliseu/presente/model/filter/ClienteFilter.java
package br.eng.eliseu.presente.model.filter;

import br.eng.eliseu.presente.model.core.BaseFilter;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ClienteFilter extends BaseFilter {
    private String nome;
    private String email;
    private String telefone;
    private Long usuarioId;
}

File: front/src/app/shared/model/filter/cliente-filter.ts
import { BaseFilter } from '../core/base-filter';

export class ClienteFilter extends BaseFilter {
  nome?: string;
  email?: string;
  telefone?: string;
  usuarioId?: number;

  constructor(init?: Partial<ClienteFilter>) {
    super();
    Object.assign(this, init);
  }
}
